packages/ui/src/components/counter-badge.tsx

```diff
@@ -0,0 +1,42 @@
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+const counterBadgeVariants = cva('badge badge-counter inline-flex w-fit items-center', {
+  variants: {
+    theme: {
+      default: 'badge-muted',
+      info: 'badge-info'
+    }
+  },
+  defaultVariants: {
+    theme: 'default'
+  }
+})
+
+type CounterBadgeProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  'color' | 'role' | 'aria-readonly' | 'tabIndex' | 'onClick'
+> & {
+  theme?: VariantProps<typeof counterBadgeVariants>['theme']
+}
+
+function CounterBadge({ className, theme = 'default', children, ...props }: CounterBadgeProps) {
```

Comment:
Since part of the requirements is to limit at 999+, I think we need to rethink this component a bit. Instead of accepting children, how about we accept a number prop called `count`? We can then sanitise the count to be a minimum of 0 and overflow 999 as `999+` with a `title` added to the `div` with the actual number.

For uses such as for with `diff`, perhaps we can add something like `showSign` where we can detect if the number is positive or negative and show a `+` or `-` and potentially show the full number.

e.g.

```typescript jsx
<CounterBadge count={1234} />
{/* renders [999+] and shows 1234 on hover */}

<CounterBadge count={-1234} showSign />
{/* renders [-1234] */}

<CounterBadge count={1234} showSign />
{/* renders [+1234] */}
```

Reply:
`CounterBadge` will undergo a few more changes, as we need to support displaying diff values in PRs, such as `+1002` (success theme) and `-2322` (danger theme). This enhancement can be incorporated into the `CounterBadge` component.

<img width="558" alt="image" src="https://github.com/user-attachments/assets/96ea842c-1bbb-4230-b1ed-e2218d6f25bd" />

@cxramina  is working on this change. Once it’s complete, we can update the component to accept a `count` prop of number type and a `truncateAtMax` prop of boolean type to handle truncation when the `count` value exceeds the maximum.